packages/kbn-es-query/src/kuery/functions/__tests__/and.js

```diff
@@ -58,6 +57,16 @@ describe('kuery functions', function () {
         );
       });
 
+      it('should pass the config to subqueries in an ES bool query\'s filter clause', function () {
+        const config = {
+          dateFormatTZ: 'America/Phoenix'
+        };
+        const node = nodeTypes.function.buildNode('and', [childNode1, childNode2]);
+        const result = and.toElasticsearchQuery(node, indexPattern, config);
+        expect(result.bool.filter).to.eql(
+          [childNode1, childNode2].map((childNode) => ast.toElasticsearchQuery(childNode, indexPattern, config))
```

Comment:
(This comment applies to the tests for `not` and `or` as well)

I don't think this is really confirming that the config gets passed along. Since the test queries do not contain date fields, the timezone will not get added to the query either way.

For this particular situation it might be better to set up a spy on the `ast.toElasticsearchQuery` method and simply assert that the config object was passed to it. Inside Mocha tests we usually use Sinon to do spys/stubs/mocks. Let me know if you'd like me to show you how to do that.

Reply:
@Bargs yes please, I'd like you to show me how to set up the spy.

Reply:
Done for ```and``` ```or``` and ```not``` method tests